cellengine/resources/gate.py

```diff
@@ -193,51 +96,177 @@ def create(
+                tailored gate, the gid of the global tailored gate must be specified.
+            locked (bool): Prevents modification of the gate via the web interface.
+            tailored_per_file (bool): Whether or not this gate is tailored per FCS file.
+            fcs_file_id (str): ID of FCS file, if tailored per file. Use ``None`` for
+                the global gate in a tailored gate group. If specified, do not
+                specify ``fcs_file``.
+            fcs_file (str): Name of FCS file, if tailored per file. An attempt
+                will be made to find the file by name. If zero or more than one file
+                exists with the name, an error will be thrown. Looking up files
+                by name is slower than using the ID, as this requires
+                additional requests to the server. If specified, do not specify
+                ``fcs_file_id``.
 
-class PolygonGate(Gate):
-    """Basic concrete class for polygon gates"""
+        Returns:
+            A RectangleGate object.
+
+        Example:
+            ```python
+            experiment.create_rectangle_gate(x_channel="FSC-A", y_channel="FSC-W",
+            name="my gate", 12.502, 95.102, 1020, 32021.2)
+            cellengine.Gate.create_rectangle_gate(experiment_id, x_channel="FSC-A",
+            y_channel="FSC-W", name="my gate", x1=12.502, x2=95.102, y1=1020,
+            y2=32021.2, gid=global_gate.gid)
+            ```
+        """
+
+        fcs_file_id = parse_fcs_file_args(
+            experiment_id,
+            tailored_per_file,
+            fcs_file_id,
+            fcs_file,
+        )
 
+        if label == []:
+            label = [numpy.mean([x1, x2]), numpy.mean([y1, y2])]
+        if not gid:
+            gid = generate_id()
+
+        model = {
+            "locked": locked,
+            "label": label,
+            "rectangle": {"x1": x1, "x2": x2, "y1": y1, "y2": y2},
+        }
+
+        return Gate(
+            id=None,
+            experiment_id=experiment_id,
+            fcs_file_id=fcs_file_id,
+            gid=gid,
+            tailored_per_file=tailored_per_file,
+            type="RectangleGate",
+            x_channel=x_channel,
+            model=model,
+            y_channel=y_channel,
+            name=name,
+        )  # type: ignore
+
+
+@define(repr=False)
+class PolygonGate(Gate):
     @classmethod
-    @doc_inherit(format_polygon_gate)
     def create(
         cls,
         experiment_id: str,
         x_channel: str,
         y_channel: str,
         name: str,
-        vertices: List[float],
-        label: List = [],
-        gid: str = None,
-        locked: bool = False,
-        parent_population_id: str = None,
-        parent_population: str = None,
-        tailored_per_file: bool = False,
-        fcs_file_id: str = None,
-        fcs_file: str = None,
-        create_population: bool = True,
+        vertices: List[List[float]],
+        label: Optional[List[float]] = [],
```

Comment:
nit for consistency with other gate classes: `Optional[]` isn't necessary here or on `locked` or `tailored_per_file` ([ref](https://docs.python.org/3/library/typing.html#typing.Optional)).